### PR TITLE
memorymapping: constrain to darwin (fails on linux)

### LINKED ISSUE #138378: memorymapping fails to build: src/fmemopen.c:62:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')
`memorymapping` fails to build as: https://hydra.nixos.org/build/153173858

```
src/fmemopen.c: In function 'seekfn':
src/fmemopen.c:62:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')
   62 |       if (offset >= 0) {
      |                  ^~
src/fmemopen.c:63:9: error: aggregate value used where an integer was expected
   63 |         pos = (size_t)offset;
      |         ^~~
src/fmemopen.c:70:18: error: invalid operands to binary >= (have 'fpos_t' and 'int')
   70 |       if (offset >= 0 || (size_t)(-offset) <= mem->pos) {
      |                  ^~
```

### Notify maintainers

@veprbl 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: memorymapping
```

### PR DESCRIPTION
**Pull request by trofi**
Closes: https://github.com/NixOS/nixpkgs/issues/138378

### CODE REVIEW

File: pkgs/development/libraries/memorymapping/default.nix [hunk 1]
```diff
@@ -37,7 +37,9 @@ stdenv.mkDerivation {
     description = "fmemopen for Mac OS and iOS";
     license = licenses.asl20;
     maintainers = with maintainers; [ veprbl ];
-    platforms = platforms.unix;
+    # Uses BSD-style funopen() to implement glibc-style fmemopen().
+    # Add more BSDs if you need to.
+    platforms = platforms.darwin;
     broken = stdenv.isAarch64;
```
Comment (veprbl) [inline pkgs/development/libraries/memorymapping/default.nix:43]: This is libc-specific so
```suggestion
    broken = stdenv.cc.isGNU && stdenv.isAarch64;
```

Comment (trofi) [reply pkgs/development/libraries/memorymapping/default.nix:43]: I suspect `isGNU` is really about `gcc/clang` compiler and not libc (it outgh to be named isGCC):

```
nix-repl> clangStdenv.cc.isGNU
false

nix-repl> stdenv.cc.isGNU
true

nix-repl> stdenv.hostPlatform.libc
"glibc"

nix-repl> clangStdenv.hostPlatform.libc
"glibc"
```

I guess what wee need is `clangStdenv.hostPlatform.libc NOT IN ("glibc", "musl")` or `clangStdenv.hostPlatform.libc IN (${BSD_VARIANTS})`.

Comment (veprbl) [reply pkgs/development/libraries/memorymapping/default.nix:43]: Fair point. So you imply that musl is also not supported? If that's the case, your original change makes more sense.

Comment (veprbl) [reply pkgs/development/libraries/memorymapping/default.nix:43]: I see it is: http://git.musl-libc.org/cgit/musl/tree/include/stdio.h?id=f368d9fd26ae002fe2fce20add4cb2b806f48972#n58

**Review by veprbl — APPROVED** [review]

**Pull request merged**